R/imageutils.R

```diff
@@ -1,19 +1,23 @@
 startPNG <- function(filename, width, height, res, ...) {
-  # shiny.useragg is an experimental option that isn't officially supported or
-  # documented. It's here in the off chance that someone really wants
-  # to use ragg (say, instead of showtext, for custom font rendering).
-  # In the next shiny release, this option will likely be superseded in
-  # favor of a fully customizable graphics device option
-  if ((getOption('shiny.useragg') %||% FALSE) && is_installed("ragg")) {
-    pngfun <- ragg::agg_png
+  # Prefer ragg over Cairo, but only consider them over a quartz (i.e., Mac)
+  # device if user opts into that behavior (if quartz isn't available,
+  # ragg/cairo are an opt-out when installed).
+  use_ragg <- getOption('shiny.useragg')
+  use_cairo <- getOption('shiny.usecairo')
+  pngfun <- if (isTRUE(use_ragg) && is_installed("ragg")) {
```

Comment:
Now that I think of it, maybe we should default to `{ragg}` (when installed) even when a quartz device available. That would make the custom fonts story a whole lot simpler (i.e., install `{ragg}` and custom fonts "just work" everywhere)